shell/platform/darwin/common/framework/Source/FlutterStandardCodec.mm

```diff
@@ -364,16 +366,22 @@ - (UInt32)readSize {
   }
 }
 
-- (NSData*)readData:(NSUInteger)length {
+- (CFDataRef)readDataRef:(NSUInteger)length {
   _range.length = length;
   NSData* data = [_data subdataWithRange:_range];
   _range.location += _range.length;
-  return data;
+  // return (CFDataRef)CFAutorelease((__bridge CFDataRef)data);
+  return (CFDataRef)CFRetain((__bridge CFDataRef)data);
```

Comment:
So `data` is already on the autorelease pool here.  You should just return it directly with the bridge cast.

`return (__bridge CFDataRef)data;`